core/silkworm/execution/processor.cpp

```diff
@@ -66,9 +67,12 @@ ValidationResult ExecutionProcessor::validate_transaction(const Transaction& txn
     return ValidationResult::kOk;
 }
 
-Receipt ExecutionProcessor::execute_transaction(const Transaction& txn) noexcept {
+void ExecutionProcessor::execute_transaction(const Transaction& txn, Receipt& receipt) noexcept {
     assert(validate_transaction(txn) == ValidationResult::kOk);
 
+    // Optimization: since receipt.logs might have some capacity, let's reuse it.
+    std::swap(receipt.logs, state_.logs());
+
     state_.clear_journal_and_substate();
 
     assert(txn.from.has_value());
```

Comment:
Maybe this should be a `SILKWORM_ASSERT` or a std::runtime error if no value

Reply:
`validate_transaction` already checks that `txn.from` is not empty, so here assert is more like documentation rather than a runtime check.

Reply:
Ok